components/application-gateway/pkg/proxy/provider/secrets_provider.go

```diff
@@ -0,0 +1,115 @@
+package provider
```

Comment:
I'm wondering whether the package names are descriptive enough. Maybe something like : /pkg/proxyconfig/provider would be better.